lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -657,9 +666,13 @@ def _publish_unit_data(
         if not host:
             host = socket.getfqdn()
 
-        unit_databag = relation.data[self.unit]
+        # TODO: what should be sent to the relation data bag if prefix is None?
```

Comment:
I think we shouldn't send anything in that case. I think pydantic should be good with accepting a `prefix=None`, only thing we need to make sure is that it doesn't get dumped to databag as a json `null`. If it does, no biggie, it will appear on the other side as None, but I think pydantic models have a config option to omit null values, and that feels a little bit cleaner.

Reply:
Which makes me realize, we need to differentiate between `None`, which means 'use the default prefix', and `""`, which means 'no prefix at all'.

It feels a bit ugly to have None mean something and an empty string mean something else. Should we perhaps go for a more complex type for `prefix`?

```
prefix=None # --> http://traefik-url/model-app-0/
prefix=Prefix(s: str) # -->  http://traefik-url/{s} 
```

This is also more future proof, may we ever need to add further config to the prefix data.

Reply:
@sed-i @simskij wdyt?

Reply:
For reference, had a discussion on this in Riga. Conclusions:

- the requirement for a 'transparent proxy' configuration of traefik is no longer applicable. 
- That configuration is possible: using "PathPrefix(`/`)" you can talk 'through' traefik and reach any endpoint behind it without prefix.
- However: doing so will effectively overwrite any of traefik's own endpoints such as /metrics and /debug. So we'll effectively lose self-monitoring.

Thus, for now we drop the 'empty' prefix configuration option and require the custom prefix to be nonempty.

The risk remains that if someone passes `prefix=metrics`, when one asks traefik for its metrics, they'll be in fact accessing (proxied application)'s metrics. 
We have to clearly document this risk and do what's possible to warn against it. Traefik charm will have to include a list of traefik's own internal endpoints such as `/api`, `/metrics`, `/debug` and so on, and warn if anyone passes a custom prefix conflicting with those.

Also, we can and should check the requested custom prefixes for duplicates: if two apps request to be exposed on `/foo`, only one of them will be. In this case the charm should probably block (@simskij ?).